src/main/java/org/dasein/cloud/digitalocean/compute/DOImage.java

```diff
@@ -323,9 +329,19 @@ private ResourceStatus toStatus(org.dasein.cloud.digitalocean.models.Image image
     @Override
     public @Nullable MachineImage getImage(@Nonnull String providerImageId) throws CloudException, InternalException {
         APITrace.begin(provider, "Image.getImage");
+        Cache<MachineImage> cache = Cache.getInstance(provider, "image-" + providerImageId, MachineImage.class, CacheLevel.CLOUD_ACCOUNT, new TimePeriod<Second>(15, TimePeriod.SECOND) );
+        Collection<MachineImage> images = ( Collection<MachineImage> ) cache.get(getContext());
+
         try {
-            Image image = (Image) getModelById(getProvider(), org.dasein.cloud.digitalocean.models.rest.DigitalOcean.IMAGE, providerImageId);
-            return toImage(image);
+            if (images != null) {
+                return images.iterator().next();
+            } else {
+                MachineImage image = toImage((Image) getModelById(getProvider(), org.dasein.cloud.digitalocean.models.rest.DigitalOcean.IMAGE, providerImageId));
+                images = new ArrayList<MachineImage>();
+                images.add(image);
+                cache.put(getContext(), images);
```

Comment:
Last three lines can be replaced with just one: `cache.put(getContext(), Collections.singletonList(image));`